services_delegate.go

```diff
@@ -171,11 +171,7 @@ func (d *servicesDelegate) NotifyJoin(node *memberlist.Node) {
 
 func (d *servicesDelegate) NotifyLeave(node *memberlist.Node) {
 	log.Debugf("NotifyLeave(): %s", node.Name)
-	go func() {
-		d.state.Lock()
-		defer d.state.Unlock()
-		d.state.ExpireServer(node.Name)
```

Comment:
Since this is the only place where `ExpireServer` is called, I moved the locks to the body of the function, to make it easier to write one of my tests. It needs to lock the state anyway.